io.c

```diff
@@ -10922,62 +10918,82 @@ pipe_pair_close(VALUE rw)
 
 /*
  *  call-seq:
- *     IO.pipe                             ->  [read_io, write_io]
- *     IO.pipe(ext_enc)                    ->  [read_io, write_io]
- *     IO.pipe("ext_enc:int_enc" [, opt])  ->  [read_io, write_io]
- *     IO.pipe(ext_enc, int_enc [, opt])   ->  [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) -> [read_io, write_io]
+ *    IO.pipe(enc_string = 'UTF-8', **opts) {|read_io, write_io] ...} -> object
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) -> [read_io, write_io]
+ *    IO.pipe(int_enc = Encoding::UTF_8, ext_enc = nil, **opts) {|read_io, write_io] ...} -> object
```

Comment:
The new version has several problems:
* it doesn't indicate that in both cases (one encoding argument, or two arguments), encodings could be strings _or_ encoding objects
* it changes places of internal and external encoding: the first one (and the only one if only it is specified) is treated as external
* I suspect it also provides misleading "default values"

I think this set of protocols should be better:
```
IO.pipe(**opts)
IO.pipe(enc, **opts)
IO.pipe(ext_enc, int_enc, **opts)
```
...and then explain that `enc` can also be one external, or `"ext:int"`
But TBH, I believe the old version described the things better (especially if we'll just replace `[, opt]` with `**opts`), even if looked unlike "Ruby method definition"

Reply:
As I said earlier, I want to do more with the encodings here; that can include examples with their effects. Can we leave that for the next PR?

Reply:
Of course, we can do it later, if in the current one we'll just reverse this part of the changes.